src/services/importService.ts

```diff
@@ -158,4 +187,74 @@ export default class ImportService implements IImportService {
         }
         return metadata;
     }
+
+    /**
+     * Generate assets for V1 Image Project frames and regions
+     * @param v1Project - v1 Project content and file information
+     * @param frameList - Dictionary of frames:regions in v1 project
+     * @param assetService - assetService corresponding to v2 project
+     */
+    private async generateImageAssets(v1Project: IFileInfo, frameList: {[frameName: string]: IV1Region[]},
+                                      assetService: AssetService): Promise<IAssetMetadata[]> {
+        const generatedAssetMetadata: IAssetMetadata[] = [];
+        const originalProject = JSON.parse(v1Project.content as string);
+
+        for (const frameName in frameList) {
+            if (frameList.hasOwnProperty(frameName)) {
+                const frameRegions = frameList[frameName];
+                const asset = AssetService.createAssetFromFilePath(
+                `${v1Project.file.path.replace(/[^\/]*$/, "")}${frameName}`);
+                const assetState = originalProject.visitedFrames.indexOf(frameName) > -1 && frameRegions.length > 0
+                ? AssetState.Tagged : (originalProject.visitedFrames.indexOf(frameName) > -1
+                ? AssetState.Visited : AssetState.NotVisited);
+                const populatedMetadata = await assetService.getAssetMetadata(asset).then((metadata) => {
+                    const taggedMetadata = this.addRegions(metadata, frameRegions);
+                    taggedMetadata.asset.state = assetState;
+                    taggedMetadata.asset.path = `file:${taggedMetadata.asset.path}`;
+                    return taggedMetadata;
+                });
+                generatedAssetMetadata.push(populatedMetadata);
+            }
+        }
+        return generatedAssetMetadata;
+    }
+
+    /**
+     * Generate assets for V1 Video Project frames and regions
+     * @param v1Project - v1 Project content and file information
+     * @param frameList - Dictionary of frames:regions in v1 project
+     * @param assetService - assetService corresponding to v2 project
+     */
+    private async generateVideoAssets(v1Project: IFileInfo, frameList: {[frameName: string]: IV1Region[]},
+                                      assetService: AssetService): Promise<IAssetMetadata[]> {
+        const generatedAssetMetadata: IAssetMetadata[] = [];
+        const parent = await this.createParentVideoAsset(v1Project);
+        const originalProject = JSON.parse(v1Project.content as string);
+
+        for (const frameName in frameList) {
```

Comment:
Consider breaking down the inner loop logic into its own function to make this easier to understand.